packages/fxa-settings/src/models/integrations/base-integration.ts

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { SyncIntegrationFeatures } from './base-sync-integration';
+
+// import { ModelDataProvider } from '../../lib/model-data';
+
+// TODO: account for 'pairing' auth-brokers
+export enum IntegrationType {
+  Base,
+  OAuthRedirect,
+  OAuthWebChannel,
+  Sync,
+  SyncChannel,
+  SyncWebChannel,
+  V3Desktop,
+  Web,
+}
+
+export abstract class Integration {
+  type: IntegrationType;
+  abstract features: Partial<IntegrationFeatures | SyncIntegrationFeatures>;
+
+  constructor(type: IntegrationType) {
+    this.type = type;
+  }
+}
+
+export type IntegrationFeatures = {
+  /**
+   * If the provided UID no longer exists on the auth server, can the
+   * user sign up/in with the same email address but a different uid?
+   */
+  allowUidChange: boolean;
+  /**
+   * Should the user agent be queried for FxA data?
+   */
+  fxaStatus: boolean;
```

Comment:
This is just a nitpick, but I like fields to be consistent in their construction. For example, if we look at the first field, `allowUidChange: boolean`. I think this is good semantics. The field name starts with a verb, `allow`, which makes sense with boolean value type. 

This `fxaStatus: boolean` field on the other hand doesn't start with a verb. Perhaps something like `hasFxaStatus` or `allowFxaStatusQuery` would be more consistent semantically. 

If we are just porting this over, it's probably not worth addressing until the code settles though.

Reply:
I thought something similarly as I was looking at these, I think it would be nice to have a ticket to review these at the end. For now, it's nice to search for the same capability name to easily see the equivalent in content-server.